src/utils/snakeToCamel.ts

```diff
@@ -0,0 +1,29 @@
+type CamelCase<T> = T extends object
+  ? {
+      [K in keyof T as Uncapitalize<CamelCaseKey<K & string>>]: CamelCase<T[K]>;
+    }
+  : T;
+
+type CamelCaseKey<K extends string> = K extends `${infer A}_${infer B}`
+  ? `${Lowercase<A>}${Capitalize<CamelCaseKey<B & string>>}`
+  : K;
+
+export function snakeToCamel<T>(obj: T): CamelCase<T> {
+  if (typeof obj !== 'object' || obj === null) {
+    return obj as CamelCase<T>;
+  }
+
+  if (Array.isArray(obj)) {
+    return obj.map((item) => snakeToCamel(item)) as CamelCase<T>;
+  }
+
+  const result = {} as CamelCase<T>;
+  for (const key in obj) {
+    if (Object.prototype.hasOwnProperty.call(obj, key)) {
```

Comment:
Do we need to use the prototype function here? because It already checked the types above
(just curious)

Reply:
@HoonBaek Where's the above?

Reply:
on the line numbers 12 and 16
Isn't already checked?

Reply:
Ahh this line is for checking if the object has the key not checking if its type is Object or Array. 

Also, this line might look unnecessary cause the `key` is from just its above like in for-in loop. 
You can find the detail of behind reason is described in [this Stackoverflow question](https://stackoverflow.com/questions/1963102/what-does-the-jslint-error-body-of-a-for-in-should-be-wrapped-in-an-if-statemen). 

But anyways, I also didn't like this pattern either. So I dropped the for-in + if pattern and replaced it \w `Object.keys.forEach(key => )`. 992def4bce33a54292532cc55703275e2c85a02a 

Thanks for mentioning it Hoon :)